fixql-core/src/main/scala/com/iterable/graphql/FromGraphQLJava.scala

```diff
@@ -0,0 +1,202 @@
+          .build())
+        .getData[java.util.LinkedHashMap[String, Any]]
+        .get(null) // the map will have a value with key = null
+
+      val valuesResolver = new ValuesResolver
+      val topLevelFields: Seq[Field.Fixed] =
+        fieldSubSelection.getMergedSelectionSet.getSubFieldsList.asScala.map { mergedField =>
+          // See DataFetchingSelectionSetImpl.traverseFields
+          val fieldDef = Introspection.getFieldDef(graphQLSchema, graphQLSchema.getQueryType, mergedField.getName)
+          val unwrappedType = GraphQLTypeUtil.unwrapAll(fieldDef.getType)
+          // See ValueFetcher.fetchValue() or other callers of DataFetchingFieldSelectionSetImpl.newCollector()
+          //val codeRegistry: GraphQLCodeRegistry = executionContext.getGraphQLSchema.getCodeRegistry
+          //val parentType: GraphQLFieldsContainer = getFieldsContainer(executionInfo)
+          //val argumentValues: util.Map[String, AnyRef] = valuesResolver.getArgumentValues(codeRegistry, fieldDef.getArguments, field.getArguments, executionContext.getVariables)
+          //val fieldDef: GraphQLFieldDefinition = executionInfo.getFieldDefinition
+          //val fieldType: GraphQLOutputType = fieldDef.getType
+          val argumentValues = valuesResolver.getArgumentValues(fieldDef.getArguments, mergedField.getArguments, context.getVariables)
+          val selectionSet = DataFetchingFieldSelectionSetImpl.newCollector(context, unwrappedType, mergedField)
+          mkField(mergedField.getName, argumentValues, selectionSet)
+        }.toSeq
+      Query[Field.Fixed](topLevelFields)
+    }
+  }
+
+  private def toJavaValues(value: Json): AnyRef = {
+    value.foldWith(new Json.Folder[AnyRef] {
+      override def onNull = null
+      override def onBoolean(value: Boolean) = value: java.lang.Boolean
+      override def onNumber(value: JsonNumber) = value.toBigDecimal
+      override def onString(value: String) = value
+      override def onArray(value: Vector[Json]) = value.map(toJavaValues)
+      override def onObject(value: JsonObject) = value.toMap.mapValues(toJavaValues)
+    })
+  }
+
+  import qq.droste.syntax.fix._
+  private def mkField(fieldName: String, arguments: java.util.Map[String, AnyRef], fss: DataFetchingFieldSelectionSet): Field.Fixed = {
+    // getFields contains flattened fields from all children, but we only want the immediate children
+    val childFields = fss.getFields.asScala.filterNot(_.getQualifiedName.contains("/"))
+    val args = PlayJson.obj(arguments.asScala.map { case (key, value) =>
+      key -> (fromJavaValue(value): play.api.libs.json.Json.JsValueWrapper)
+    }.toSeq: _*)
+    Field[Field.Fixed](
+      fieldName,
+      args,
+      childFields.map { selectedField =>
+        val subfss = selectedField.getSelectionSet
+        mkField(selectedField.getName, selectedField.getArguments, subfss)
+      }.toSeq,
+    ).fix
+  }
+
+  private def fromJavaValue(value: Any): JsValue = {
+    value match {
```

Comment:
We don't ever have other types here? like `Double`? Is that just how the Java GraphQL library works.

In any case I usually prefer to add a default case like:

```scala
case v => throw new IllegalArgumentException(s"Java value $v has unexpected type ${v.getClass}")
```

Reply:
will do